app/actions/GovernanceActions.js

```diff
@@ -153,41 +153,47 @@ export const compareInventory = () => async (dispatch, getState) => {
   try {
     const oldInventory = sel.inventory(getState());
     const piURL = sel.politeiaURL(getState());
-    let oldProposals = sel.proposals(getState());
-    const keys = Object.keys(oldInventory);
+    const oldProposals = sel.proposals(getState());
+    const newProposalsList = getDefaultInventory();
 
     const { data } = await pi.getTokenInventory(piURL);
     const inventory = updateInventoryFromApiData(data);
 
     let isDifferent = false;
 
     // Check if old proposals list have had some proposal removed from inventory and
-    // also remove it from the list.
-    for (let i = 0; i < keys.length; i++) {
-      const key = keys[i];
-      oldProposals[key].forEach( (proposal, index) => {
-        if (!inventory[key].includes(proposal.token)) {
-          oldProposals[key].splice(index, 1);
+    // also remove it from proposal's list.
+    Object.keys(oldProposals).forEach( key => {
+      newProposalsList[key] = oldProposals[key].reduce((acc, p) => {
+        if(inventory[key].includes(p.token)) {
+          acc.push(p);
+        } else {
+          // Remove token from old inventory so we can re-fetch it.
+          oldInventory[key].splice(oldInventory[key].indexOf(p.token), 1)
           isDifferent = true;
         }
-      });
-
-      if (isDifferent) {
-        isDifferent = false;
-        dispatch({ type: REMOVED_PROPOSALS_FROM_LIST, proposals: oldProposals });
-      }
+        return acc;
+      }, []);
+    });
+    if (isDifferent) {
+      dispatch({ type: REMOVED_PROPOSALS_FROM_LIST, proposals: newProposalsList });
     }
 
-    // Compare difference betweeen old inventory and inventory we just get and
-    // update proposal list with missing proposals.
-    for (let i = 0; i < keys.length; i++) {
-      const key = keys[i];
-      const diffHashes = inventory[key].filter( token => !oldInventory[key].includes(token) && (isDifferent = true));
+    // create array with all old inventory and inventory token's values
+    const flatOldProps = [oldInventory.activeVote, oldInventory.abandonedVote, oldInventory.finishedVote, oldInventory.preVote].reduce( (acc, v) => {
+      v.forEach(vp => acc.push(vp));
+      return acc;
+    }, []);
+    const flatNewProps = [inventory.activeVote, inventory.abandonedVote, inventory.finishedVote, inventory.preVote].reduce((acc, v) => {
+      v.forEach(vp => acc.push(vp));
+      return acc
+    }, []);
 
-      if (isDifferent) {
-        isDifferent = false;
-        dispatch(getProposalsAndUpdateVoteStatus(diffHashes));
-      }
+    // Get difference between new inventory and old one, so we can bring a batch
+    // of new proposals.
+    const diffHashes = flatNewProps.filter( token => !flatOldProps.includes(token));
```

Comment:
This is fine for the current number of props but bear in mind this is an O(n^2) algo. It would be better to convert `flatOldProps` to a hashmap so you can directly check for element inclusion (instead of using `includes`) or sort both arrays and check them sequentially.

Reply:
Changed for map instead